concepts/pointers/about.md

```diff
@@ -0,0 +1,404 @@
+If the type is four bytes in length, the second element will be four bytes higher than the first element.
+
+This is also why pointers are declared to be of a certain type.
+If a pointer did not know what type it points to, it would not know how many bytes to advance when adding `1` to it.
+Adding `1` to a pointer does not necessariy mean we're adding one byte to the address.
+It means we're adding `1` times the length of its type in bytes.
+
+But wait, there's more! You didn't think we were finished with `*`, did you?
+
+## Dereference Operator
+
+Now that we understand how addresses work with array indexing, what good is it if we don't get the value stored at an address?
+How do we do that?
+To get the value stored at an address we use the dereference operator, which is, you guessed it, `*`.
+
+Come on!
+Are you kidding me?
+How many uses are there for `*`?!
+Multiplication, declaring a pointer, and now derefencing!
+It's confusing!
+And what exactly is dereferencing, anyway?
+
+An example shows how to use the dereferencing operator
+
+```c
+#include <stdio.h>
+
+int main() {
+    // define the variable
+    int my_ints[] = {1, 2};
+    
+    // define the pointer from the address of the variable
+    int *my_ptr = my_ints;
+    
+    // prints: my_ints[0] is 1. *(my_ptr + 0) is 1.
+    printf("my_ints[0] is %u. *(my_ptr + 0) is %u.\n", my_ints[0], *(my_ptr + 0));
+    
+    // prints: my_ints[1] is 2. *(my_ptr + 1) is 2.
+    printf("my_ints[1] is %u. *(my_ptr + 1) is %u.\n", my_ints[1], *(my_ptr + 1));
+    
+    return 0;
+}
+```
+
+Using parentheses can help remove ambiguity of intentions when using pointer arthimetic.
+The following example may not print what the programmer intended to
+
+```c
+#include <stdio.h>
+
+int main() {
+    int my_ints[] = {1, 42};
+    int *my_ptr = my_ints;
+    
+    // prints: my_ints[0] is 1. *++my_ptr is 42.
+    printf("my_ints[0] is %u. *++my_ptr is %u.\n", my_ints[0], *++my_ptr);
+    
+    // prints: my_ints[1] is 43. ++*my_ptr++ is 43.
+    printf("my_ints[1] is %u. ++*my_ptr++ is %u.\n", my_ints[1], ++*my_ptr);
+
+    return 0;
+}
+```
+
+Both the `*` and `++` have the same precedence, but they evaluate from right to left.
+In the first line `my_ptr` is first incremented to the second element, which is then dereferenced.
+In the second line `my_ptr`, which was incremented to the second element in the first line, is first dereferenced, and then the dereferenced value is incremented.
+If we add this line to the above example
+
+```c
+// prints: my_ints[1] is 43. *++my_ptr is (some garbage value).
+printf("my_ints[1] is %u. *++my_ptr is %u.\n", my_ints[1], *++my_ptr);
+```
+
+In that line we have incremented the pointer to be beyond the array.
+We have overrun the array and we are pointing at the next four bytes in memory which the pointer interprets as an int, but, in fact, those bytes could belong to another variable or variables.
```

Comment:
Maybe it's only me being pedantic, but I would treat "int" as C code (the English translation being "integer"), just the same as you have done later on with "`char`s" (the English translation being "characters").

```suggestion
We have overrun the array and we are pointing at the next four bytes in memory which the pointer interprets as an `int`, but, in fact, those bytes could belong to another variable or variables.
```